CTOR-1153 : Refresh cloud-cloudfoundry-api (#3848)
centreon/centreon-documentation: pp/integrations/plugin-packs/procedures/cloud-cloudfoundry-api.md

---
id: cloud-cloudfoundry-api
title: Cloud Foundry API
---
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

## Connector dependencies

> The following monitoring connectors will be installed when you install the **Cloud Foundry** connector through the

Reviewer: ```suggestion
The following monitoring connectors will be installed when you install the **Cloud Foundry API** connector through the
```

**Configuration > Monitoring Connector Manager** menu:
* [Base Pack](./base-generic.md)

## Pack assets

### Templates

> The Monitoring Connector **Cloud Foundry** brings a host template:

Reviewer: ```suggestion
The Monitoring Connector **Cloud Foundry API** brings a host template:
```

* **Cloud-Cloudfoundry-Api-custom**

The connector brings the following service templates (sorted by the host template they are attached to):

<Tabs groupId="sync">
<TabItem value="Cloud-Cloudfoundry-Api-custom" label="Cloud-Cloudfoundry-Api-custom">

| Service Alias   | Service Template                              | Service Description      |
|:----------------|:----------------------------------------------|:-------------------------|

> | Apps-State      | Cloud-Cloudfoundry-Apps-State-Api-custom      | Check applications state |

Reviewer: ```suggestion
| Apps-State      | Cloud-Cloudfoundry-Apps-State-Api-custom      | Check the state of applications |
```

> | Instances-State | Cloud-Cloudfoundry-Instances-State-Api-custom | Check instances state    |

Reviewer: ```suggestion
| Instances-State | Cloud-Cloudfoundry-Instances-State-Api-custom | Check the state of instances |
```

> The services listed above are created automatically when the **Cloud-Cloudfoundry-Api-custom** host template is used.

</TabItem>
</Tabs>

### Collected metrics & status

Here is the list of services for this connector, detailing all metrics and statuses linked to each service.

<Tabs groupId="sync">
<TabItem value="Apps-State" label="Apps-State">

| Name                       | Unit  |
|:---------------------------|:------|
| applications.started.count | count |
| applications.stopped.count | count |
| state                      | N/A   |
| state                      | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
<TabItem value="Instances-State" label="Instances-State">

| Name                    | Unit  |
|:------------------------|:------|
| app-state               | N/A   |
| instances.running.count | count |
| instances.stopped.count | count |
| instances.crashed.count | count |
| instance-state          | N/A   |

> To obtain this new metric format, include **--use-new-perfdata** in the **EXTRAOPTIONS** service macro.

</TabItem>
</Tabs>

## Prerequisites

> Your Centreon poller must be able to access your Cloud Foundry instance (have its hosting URL).
> You must have a Cloud Foundry user account with the necessary permissions to access the data monitored via the API (applications, etc.).
> Ensure that the Centreon server can communicate with the Cloud Foundry API over the network.
> Allow outbound access to the API URL (https://api./<domain/>) through the required ports (typically 443 for HTTPS).
> Verify that the Cloud Foundry API's SSL certificate is valid and, if necessary, add it to the trusted certificates of the Centreon poller.
> For more informations, refers to the [official documentation](https://docs.cloudfoundry.org/)

Reviewer: ```suggestion
* Your Centreon poller must be able to access your Cloud Foundry instance (have its hosting URL ready).
* You must have a Cloud Foundry user account with the necessary permissions to access the data monitored via the API (applications, etc.).
* Ensure that the Centreon server can communicate with the Cloud Foundry API over the network.
* Allow outbound access to the API URL (https://api./<domain/>) through the required ports (typically 443 for HTTPS).
* Check that the Cloud Foundry API's SSL certificate is valid and, if necessary, add it to the trusted certificates of the Centreon poller.
* For more information, refer to the [official Cloud Foundry documentation](https://docs.cloudfoundry.org/)
```

| EXTRAOPTIONS          | Any extra option you may want to add to the command (a --verbose flag for example). All options are listed [here](#available-options). | --verbose              |             |

</TabItem>
</Tabs>

3. [Deploy the configuration](/docs/monitoring/monitoring-servers/deploying-a-configuration). The service appears in the list of services, and on the **Resources Status** page. The command that is sent by the connector is displayed in the details panel of the service: it shows the values of the macros.

## How to check in the CLI that the configuration is OK and what are the main options for?

Once the plugin is installed, log into your Centreon poller's CLI using the
**centreon-engine** user account (`su - centreon-engine`). Test that the connector 
is able to monitor a resource using a command like this one (replace the sample values by yours):

```bash
/usr/lib/centreon/plugins/centreon_cloudfoundry_restapi.pl \
	--plugin=cloud::cloudfoundry::restapi::plugin \
	--mode=apps-state \
	--custommode='restapi' \

> 	--hostname='' \

Reviewer: hostname

	--api-path='/v2' \
	--api-username='' \

> 	--api-password='' \

Reviewer: username and password

> 	--hostname='' \
> 	--api-path='/v2' \
> 	--api-username='' \
> 	--api-password='' \

Author: ```suggestion
	--hostname='10.0.0.1' \
	--api-path='/v2' \
	--api-username='XXXX' \
	--api-password='XXXX' \
```
